src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -208,12 +208,18 @@ public async ValueTask SetupAsync()
                 BinaryPrimitives.WriteUInt32BigEndian(_outgoingBuffer.AvailableSpan, windowUpdateAmount);
                 _outgoingBuffer.Commit(4);
 
-                await _stream.WriteAsync(_outgoingBuffer.ActiveMemory).ConfigureAwait(false);
+                await _stream.WriteAsync(_outgoingBuffer.ActiveMemory, cancellationToken).ConfigureAwait(false);
                 _rttEstimator.OnInitialSettingsSent();
                 _outgoingBuffer.Discard(_outgoingBuffer.ActiveLength);
 
                 _expectingSettingsAck = true;
             }
+            catch (OperationCanceledException oce) when (oce.CancellationToken == cancellationToken)
```

Comment:
It might be cleaner to have just a single catch block (both in this case and the later case), e.g.
```C#
catch (Exception e)
{
    Dispose();

    if (e is OperationCanceledException oce && oce.CancellationToken == cancellationToken)
    {
        // AddHttp2ConnectionAsync handles this OCE separately so don't wrap it.
        throw;
    }

    throw new IOException(SR.net_http_http2_connection_not_established, e);
}
```